### PR TITLE
Updating and including content on the PSF introduction

### PR DESCRIPTION
**Pull request by tobias-liaudat**
This PR updates and includes content of the PSF introduction.

Comments and suggestions are welcomed!

### CODE REVIEW

File: shearbook/psf/1-intro/definitions.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Definition
 
-We can vaguely define the Point Spread Function as the response of imaging system to a point source. The concept of PSF is used throughout many imaging applications, some of them are astronomical imaging and medical imaging. Depending on the science application the definition and its sources may vary. However, they all include the [diffraction](https://en.wikipedia.org/wiki/Diffraction) phenomena. The fact of having a non-infinite lens sets a limit on the resolution of the imaging system.
+We can vaguely define the Point Spread Function (PSF) as the response of imaging system to a point source. The concept of PSF is used throughout many imaging applications, some of them are astronomical imaging and medical imaging. Depending on the science application the definition and its sources may vary. However, they all include the [diffraction](https://en.wikipedia.org/wiki/Diffraction) phenomena. The fact of having a non-infinite lens sets a limit on the resolution of the imaging system.
```
Comment (aguinot) [inline shearbook/psf/1-intro/definitions.md:3]: - "We can vaguely define": there is probably a better way to say that.

File: shearbook/psf/1-intro/definitions.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
 
 We will focus on PSFs for astronomical imaging and analyze its main sources.
+
+
+## Types of variations
+
+- __Spatial__
+
+The variations that change as a function of the focal plane positions. These are main type of variations targeted by the PSF models. They mainly depend on the telescope's optics and the spatial correlations introduced by the atmosphere.
```
Comment (aguinot) [inline shearbook/psf/1-intro/definitions.md:12]: Just a suggestion: "The PSF varie as a function of the position on the focal plane. These variations are the main...."
Also, it really depends of the kind of science you want to do. In some cases the temporal variations are the main ones (adaptive optics, people working on speckles,...). You should precise for "long" exposure time on large focal plane (if your focal plane is an optical fiber you don't really care about spatial variations).
Maybe precise "turbulence in the atmosphere". If the atmosphere is static it wouldn't be a big problem.

File: shearbook/psf/1-intro/definitions.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+
+- __Spectral__
+
+The PSF is a function of the wavelength. The PSF at a specific wavelength is usually called a monochromatic PSF. The observations of our telescope are integrated at all the wavelengths allowed by the filter's bandpass used. As a consequence, our observations are polychromatic. Depending if the filter's bandpass is narrow or large and the given error requirements, the modelling of the PSF spectral variations might be required (e.g. [Euclid](https://www.euclid-ec.org)) or can be neglected (e.g. [DES](https://www.darkenergysurvey.org)).
```
Comment (aguinot) [inline shearbook/psf/1-intro/definitions.md:16]: I think that I would rephrase a bot this paragraph. I see your point and I agree with it but I feel the explanation unclear. Maybe one sentence on narrow filter like DES where the spectral variation can be neglected. Then, an example of large band pass like Euclid where it has to be included in the PSF modeling. Also, you should define "narrow" and "large" filters.
You could also mention DCR (not 100% sure of the acronym). Basically it is the diffraction due to the atmosphere which also has a chromatic dependence. Normally it is not a problem with narrow filters if the target is high in the sky (less atmosphere to go through) but it can be an issue if the target is close to the horizon. This problem can rise even with filters considered as narrow (for example we will probably have it in UNIONS for observations around the north galactic cap since our r-band filter is not that "narrow")

File: shearbook/psf/1-intro/definitions.md [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+
+- __Temporal__
+
+The PSF also varies temporally. The time we are integrating the PSF is given by the exposure time. This is a fundamental parameter that will determine several characteristics of the PSF. For example, for ground-based surveys, it changes the impact of the atmosphere on the PSF. 
+The instruments characteristics also change over time giving the PSF a low frequency variation. For space-based surveys where the atmosphere does not play a role, the characteristics of the instrument change over time following temperature changes for example. This makes unpractical the estimation of a single PSF model for all the exposures, as all the instrument's variability over time and its influence over the PSF should be modelled. It is common to estimate PSF models for each exposure (or several exposures) in order to tackle this low frequency temporal variations.
```
Comment (aguinot) [inline shearbook/psf/1-intro/definitions.md:21]: - "following temperature changes": suggestion "following temperature fluctuations". You can also say "(due to the orientation in regard to the Sun)" something like that. Those things are suppose to be "absorb" by the telescope and computed oil advance but nothing is perfect ;).
- "This makes unpractical the estimation of a single PSF model" I don't think we would ever do that. But I think what you mean is having a PSF determine in a lab under "perfect" conditions. Usually the model is parametrize and adjusted on observed stars for each exposures.

File: shearbook/psf/1-intro/contributions.md [hunk 5]
```diff
@@ -4,8 +4,20 @@
 ## Optic level
 
 - __Mirrors__
+
+    - _Size_
+
+    The size of the main mirror has a direct implicance to the size of the PSF. Following the diffraction theory, the larger the mirror for the lens, the smaller size the PSF should have. There are practical limitations on the maximal sizes of mirrors that are due to the manufacturing process. As an example, we can show the impact of the telescope's pupil diameter in the size of the PSF in the ideal Airy PSF we only take into account the diffraction phenomena. For this type of PSF we can write the Full Width Half Maximum (FWHM) as a function of the wavelength $\lambda$ and the pupil diameter $D$ as,
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:10]: Some people could argue that we are building telescope with a diameter of 40m. It might interesting to talk about segmented mirrors since we are going to see more and more of them. But they are not a miracle solution since the PSF on those mirrors is kind of a mess..

File: shearbook/psf/1-intro/contributions.md [hunk 6]
```diff
@@ -4,8 +4,20 @@
...
+
+    $$
+    \text{FWHM}_{\text{Airy}} = 1.028 \; \frac{\lambda}{D} .
+    $$
+
+
     - _Aberrations_
-        - There are imperfections in the curvature of mirrors producing different types of aberrations. These introduce optical path differences for the incoming light rays producing features in the PSF. 
+    
+    There are imperfections in the curvature of mirrors producing different types of aberrations. These introduce optical path differences for the incoming light rays producing features in the PSF. They are usually modelled using [Zernike polynomials](https://en.wikipedia.org/wiki/Zernike_polynomials) which are a sequence of polynomials that are orthogonal in the unit disk.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:19]: I am not sure to understand " which are a sequence of polynomials that are orthogonal in the unit disk" some details on that might be nice.

File: shearbook/psf/1-intro/contributions.md [hunk 7]
```diff
@@ -26,10 +38,10 @@ Typical aberrations with simulated in-focus stars with typical seeing. Credit: A
 ```
 
 - __Mirrors__
-    - _Size_
-        - The size of the main mirror has a direct implicance to the size of the PSF. Following the diffraction theory, the larger the mirror for the lens, the smaller size the PSF should have. There are practical limitations on the maximal sizes of mirrors that are due to the manufacturing process. 
+
     - _Polishing effects_
-        - The polishing of the mirrors aims to have a perfectly smooth surface. However, the manufacturing process is not perfect and small errors remain in the surface of the mirrors. They are sometimes referred to as _surface errors_.
+    
+    The polishing of the mirrors aims to have a perfectly smooth surface. However, the manufacturing process is not perfect and small errors remain in the surface of the mirrors. They are sometimes referred to as _surface errors_.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:44]: I think you can just say: "They are sometimes referred to _surface errors_." without "as".

File: shearbook/psf/1-intro/contributions.md [hunk 8]
```diff
@@ -55,14 +68,21 @@ Simulated focal plane of LSST. We used the CCD assembly/fabrication specificatio
...
     - _Wavelength dependency_
-        - The optical system is composed by several elements where some of them can be non-ideal refractive elements. This means that they introdcuce a wavelength dependance variation.
+    
+    The optical system is composed by several elements where some of them can be non-ideal refractive elements. This means that they introdcuce a wavelength dependance variation.
+
+
     - _Obscurations_
-        - Depending on the optical system design it might require arms to sustain the camera or even superposing mirrors that block each other. These obscurations modify the shape of the PSF and vary depending on the source's angle of incidence. They are 3D structures that are projected on the 2D pupil plane.
+
+    Depending on the optical system design it might require arms to sustain the camera or even superposing mirrors that block each other. These obscurations modify the shape of the PSF and vary depending on the source's angle of incidence. They are 3D structures that are projected on the 2D pupil plane. As the obscurations are projected into the pupil plane they vary as a function of the angle of incidence.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:82]: - "or even superposing mirrors that block each other": this sounds weird to me.. I wouldn't use "superposing" and "block". You talk about secondary or third mirrors which on the line of sight.
- Also, there is two kinds of obscurations here. The one from the other mirrors/camera and the one from the arms. You should clearly separate them to make it clear that both are a "problem".

File: shearbook/psf/1-intro/contributions.md [hunk 9]
```diff
@@ -55,14 +68,21 @@ Simulated focal plane of LSST. We used the CCD assembly/fabrication specificatio
...
+
+    Depending on the optical system design it might require arms to sustain the camera or even superposing mirrors that block each other. These obscurations modify the shape of the PSF and vary depending on the source's angle of incidence. They are 3D structures that are projected on the 2D pupil plane. As the obscurations are projected into the pupil plane they vary as a function of the angle of incidence. 
+    
+    The obscurations of the pupil plane have a direct relation with the shape of the PSF, specially for space-based surveys where the PSFs are closer to the diffraction limit. The [Fraunhoffer approximation](https://en.wikipedia.org/wiki/Fraunhofer_diffraction) of diffraction is most of the time a valid approximation for astronomical telescopes. This approximation allows to relate the wavefront at the pupil plane and the observed PSF with the Fourier transform. The obscurations force a zero amplitude value for the wavefront at the obscured parts of the pupil plane. The Fourier transform of these zero values can be easily identified in diffraction-limited PSFs, being an important contributor to the PSF shape.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:84]: - "astronomical telescope": this sounds weird to me. What do you mean?

File: shearbook/psf/1-intro/contributions.md [hunk 10]
```diff
@@ -75,7 +95,9 @@ Projected obscurations of the Hubble Space Telescope. Credit: {cite}`krist2011`.
...
+    
+    The filters serve to select the wavelengths incoming to the detector. Ideally they would be a perfect band-pass filter, begin one at the desired wavelength range and zero on the other wavelengths. However, it is not possible to manufacture an ideal filter. A real one does not have an abrupt transition, meaning that there are out-of-band contributions. It is also non-flat, meanining that the passband is near the unity but it changes with the wavelength. See this [page](https://www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/en/megapipe/docs/filt.html) to visualise some examples of filters. As mentioned, the observations are always polychromatic, meaning that they are integrated in wavelength in the passband of the filter being used.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:99]: - "begin": do you mean "being"?
- Also there is not a direct link to the PSF in this paragraph. Maybe say something like "the PSF will always be polychromatic in some extend but in most cases this is not a problem because this dependance is negligible for most of the science people do".

File: shearbook/psf/1-intro/contributions.md [hunk 11]
```diff
@@ -89,8 +111,24 @@ MegaCam filter set transmission for broad band filters u, g, r, i and z. Credit:
 
 ## Ground-based
 
+- __Seeing__
+
+    In astronomy, seeing refers to the degradation of the image of an astronomical object due to the Earth's atmosphere. It is measured using three common descriptors: i) the Full Width Half Maximum (FWHM) of the PSF, ii) the Fried parameter $r_0$ and the atmospheric time constant $t_0$, and iii) the profile of the turblence strength as a function of the altitude $C_{N}^{2}$. The Fried parameter is a measure of the quality of optical transmission through the atmosphere due to random inhomogeneities in the atmosphere's refractive index.
+    In terms of FWHM, if the seeing is above $1.5$ arcsec, the observations are practically unusable for weak-lensing purposes. The best sites on Earth can reach up to $0.4$ arcsec.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:117]: Just so you know, it can be even better than that at the CFHT. Apparently 20% of the time it can reach space quality (almost). (no need to put that in the text)

File: shearbook/psf/1-intro/contributions.md [hunk 12]
```diff
@@ -89,8 +111,24 @@ MegaCam filter set transmission for broad band filters u, g, r, i and z. Credit:
 
 ## Ground-based
 
+- __Seeing__
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:114]: Why separate seeing and atmosphere?

File: shearbook/psf/1-intro/contributions.md [hunk 13]
```diff
@@ -126,18 +165,24 @@ Frame of the [video](https://www.youtube.com/watch?v=HU1i6JzvIzY) that shows a G
...
+    
 - __Thermal variations__
-    - The satellite experiences time where the sun is closer or were it is eclipsed by other object. This can occasionate high temperature changes in the space telescope. These temperature variations dilate and contract the optical system making it change of state as a function of temperature. It is sometimes to as _telescope's breathing_ for its repetitive pattern due to the orbits. See {cite}`nino2007` for an HST study of temperature variations.
+
+    The satellite experiences time where the sun is closer or were it is eclipsed by other object. This can occasionate high temperature changes in the space telescope. These temperature variations dilate and contract the optical system making it change of state as a function of temperature. It is sometimes to as _telescope's breathing_ for its repetitive pattern due to the orbits. See {cite}`nino2007` for an HST study of temperature variations.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:173]: - "It is sometimes to as": I think there is some words missing here.

File: shearbook/psf/1-intro/contributions.md [hunk 14]
```diff
@@ -126,18 +165,24 @@ Frame of the [video](https://www.youtube.com/watch?v=HU1i6JzvIzY) that shows a G
 ## Space-based
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:165]: I agree with the different point in this section but they are also true for ground base telescope. The guiding is more or less the same (probably easier on the ground) and the thermal variations are an issue for ground based telescope.

File: shearbook/psf/1-intro/contributions.md [hunk 15]
```diff
@@ -126,18 +165,24 @@ Frame of the [video](https://www.youtube.com/watch?v=HU1i6JzvIzY) that shows a G
...
 - __Polarisation__
-    - Starlight can be naturally polarised by galactic foreground dust. The different polarisations that the incoming light has can interact differently with refractive elements of the optical system. This occasionates variations that are polarisation dependent. This source is currently being studied. 
+
+    Starlight can be naturally polarised by galactic foreground dust. The different polarisations that the incoming light has can interact differently with refractive elements of the optical system. This occasionates variations that are polarisation dependent. This source is currently being studied. 
 
 
 ## Detector level
 
 
 - __Undersampling and pixelation__
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:183]: Isn't it "pixelisation"?

File: shearbook/psf/1-intro/contributions.md [hunk 16]
```diff
@@ -162,13 +208,18 @@ Surface fits to the measured WFC charge diffusion blur kernel widths shown over
...
+
 - __Brighter Fatter Effect__
-    - Each pixel in an camera is not independent of each other. One charge starts accumulating it modifies boundary pixel's electric fields. This occasionates a variation of the PSF size as a function of the flux and the wavelength. See {cite}`guyonnet2015` for a study and a model of the effect.
+
+    Each pixel in an camera is not independent of each other. One charge starts accumulating it modifies boundary pixel's electric fields. This occasionates a variation of the PSF size as a function of the flux and the wavelength. See {cite}`guyonnet2015` for a study and a model of the effect.
+
 - __Quantum efficiency__
-    - This effect describes the ratio of created electrons from the number of incoming photons. It depends on the wavelength and on the detector's technology.
+    This effect describes the ratio of created electrons from the number of incoming photons. It depends on the wavelength and on the detector's technology.
```
Comment (aguinot) [inline shearbook/psf/1-intro/contributions.md:219]: Maybe mention that for current detectors this is very good.

**Review by aguinot — COMMENTED** [review]
For some reason I didn't manage to properly compile your branch. The modifications didn't appear.. It might just be me.

**Comment by aboucaud** [comment]
@aguinot Have you tried `jupyter-book clean shearbook/` before a new build ?

File: shearbook/psf/1-intro/contributions.md (hunk 6 above)
Comment (aboucaud) [reply shearbook/psf/1-intro/contributions.md:19]: suggestion : "[...] a series of polynomials that form an orthogonal basis on which the PSF can be decomposed (in wavefront space)"

File: shearbook/psf/1-intro/contributions.md (hunk 7 above)
Comment (aboucaud) [reply shearbook/psf/1-intro/contributions.md:44]: Nope, this is an English construct : _to refer to as_

Comment (aguinot) [reply shearbook/psf/1-intro/contributions.md:44]: Ok, that's my bad. Wasn't ure about it..

**Comment by aguinot** [comment]
> @aguinot Have you tried `jupyter-book clean shearbook/` before a new build ?

I did. But it didn't work.

File: shearbook/psf/1-intro/definitions.md (hunk 1 above)
Comment (thuiop) [reply shearbook/psf/1-intro/definitions.md:3]: Just removing the "vaguely" may be enough.

File: shearbook/psf/1-intro/contributions.md (hunk 9 above)
Comment (thuiop) [reply shearbook/psf/1-intro/contributions.md:84]: I think he means by opposition to other optical systems ; this approximation requires the source to be very far from the system which is always the case for astronomy. I think replacing "for astronomical telescopes" by "in astronomy" would be better. Also, it is written "Fraunhofer" (with a single f)
